flow/project.py

```diff
@@ -2608,6 +2608,7 @@ def submit(self, bundle_size=1, jobs=None, names=None, num=None, parallel=False,
 
         # Gather all pending operations.
         default_directives = self._get_default_directives()
+        names = args.operation_name if args.operation_name else None
```

Comment:
Why are you using `args` here? The names are already unpacked in the function parameter list.

Reply:
I see where this came from in `_main_submit`. You should just be able to drop this entirely, it's getting unpacked by passing the args through to `submit`.

Reply:
oh okay, this is what I was asking about the other day. thanks for clarifying!

Reply:
i moved this back to _main_submit. names has to be explicitly passed in because it's named `operation_name` in args